src/main/java/net/snowflake/client/core/FileCacheManager.java

```diff
@@ -139,6 +139,14 @@ FileCacheManager build()
       {
         // use tmp dir if not exists.
         homeDir = systemGetProperty("java.io.tmpdir");
+      } else 
+      {
+        // Checking if home directory is writable.
+        File homeFile = new File(homeDir);
+        if (!homeFile.canWrite()) {
```

Comment:
It would be an unusual situation but you also need to be able to read. I would also make sure that the file cache is disabled if you run out of disk space.